### ISSUE TITLE
can't interpolate artifact"..." into cmd literal

### ISSUE THREAD

**Issue by simeonschaub** [JuliaLang/Pkg.jl#1954]
On latest Julia master:
```julia
julia> using Pkg.Artifacts

julia> artifact"JuliaMono"
"/home/simeon/.julia/artifacts/65279f9c8a3dd1e2cb654fdedbe8cd58889ae1bc"

julia> `$(artifact"JuliaMono")`
ERROR: LoadError: Cannot locate '(Julia)Artifacts.toml' file when attempting to use artifact 'JuliaMono' in 'Main'
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] @artifact_str(__source__::LineNumberNode, __module__::Module, name::Any)
   @ Pkg.Artifacts /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Artifacts.jl:1051
in expression starting at none:1

To edit a specific method, type the corresponding number into the REPL and press Ctrl+Q
```
Perhaps this is even an upstream Julia issue? Or is the issue that the `@artifact_string` macro is expanded in a different module? Weirdly enough, the following also works fine:
```julia
julia> "$(artifact"JuliaMono")"
"/home/simeon/.julia/artifacts/65279f9c8a3dd1e2cb654fdedbe8cd58889ae1bc"
```

**Comment by simeonschaub** [comment]
See JuliaLang/Julia#37005

**Comment by simeonschaub** [comment]
Fixed by JuliaLang/Julia#37007.
